doc/2/getting-started/flutter/index.md

```diff
@@ -0,0 +1,138 @@
+
+# Getting Started with Kuzzle and Flutter
+
+This section deals with **Kuzzle V2** (+ **Dart SDK 2**) and **VueJS**. We will create **documents** in Kuzzle and subscribe to [document notifications](/sdk/dart/2/essentials/realtime-notifications#document-messages) to develop a realtime chat.
+
+## Requirements
+
+- **Running Kuzzle V2 Stack** ([instructions here](/core/2/guides/getting-started/running-kuzzle))
+- **Setup an editor** ([instructions here](https://flutter.dev/docs/get-started/editor))
+- **Create a new project** ([instructions here](https://flutter.dev/docs/get-started/test-drive?tab=androidstudio#create-app))
+
+## Use the Kuzzle SDK in your flutter app
+
+You can find the kuzzle package on pub.dev. ([https://pub.dev/packages/kuzzle](https://pub.dev/packages/kuzzle))
+
+The `pubspec.yaml` file manages the assets and dependencies for a Flutter app. In `pubspec.yaml`, add kuzzle (2.0.1 or higher) to the dependencies list:
+
+```yaml
+dependencies:
+  flutter:
+    sdk: flutter
+  kuzzle: ^2.0.1
+```
+
+Then run `flutter pub get` to install the new dependency
+
+```bash
+$ flutter pub get
+```
+
+## App entry point
+
+Let's start with the main:
+
+_main.dart:_
+
+<<< ./snippets/main.dart.snippet[dart]
+
+## Set a username
+
+Here we will ask the user to enter its username.
+To keep it simple, this does not use the authentication system of Kuzzle but feel free to implement it for a better usage.
+
+_login.dart:_
+
+<<< ./snippets/login.dart.snippet[dart]
+
+## The chat
+
+We have to connect the server so that our client can interact with it (note that the host is the API of an android emulator in this example).
+
+In our _chat.dart_ let's import the sdk:
+
+<<< ./snippets/chat.dart.snippet:1[dart]
+
+Then we will [establish the connection](/sdk/dart/2/core-classes/kuzzle/connect) to kuzzle and create, if they don't [exist](/sdk/dart/2/controllers/index/exists/), the [index](/sdk/dart/2/controllers/index/create) and [collection](/sdk/dart/2/controllers/collection/create) of our chat, as well as fetching messages sorted by creation date and also subscribing to the same collection to receive new message in realtime.
+
+### Create index and collection
+
+So first let's write a method which will create the index/collection if it does not exists:
+
+<<< ./snippets/chat.dart.snippet:2[dart]
+
+
+### Get existing messages
+
+The method _fetchMessage()_ which will [search](/sdk/dart/2/controllers/document/search) for the first hundred newest messages.
+It will then update the state of the widget to store those messages in the `messages` variable.
+
+<<< ./snippets/chat.dart.snippet:3[dart]
+
+### Receive new messages in realtime
+
+And finally the method to subscribe to our collection.
+It will call the Kuzzle's realtime controller to allow us to receive [real-time notifications](/core/2/api/essentials/notifications) on message creations, then add it to our list of previously fetched `messages` to render it in our list by updating the state.
```

Comment:
```suggestion
It will call the Kuzzle's realtime controller to allow us to receive [real-time notifications](/core/2/api/essentials/notifications) on message creations. New messages received that way will then be added to our list of previously fetched `messages`, and rendered in our list, by updating the state.
```